cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -1196,6 +1198,32 @@ var _ = Describe("Operation Controller", func() {
 				err = k8sClient.Update(ctx, applicationCR)
 				Expect(err).To(BeNil())
 			}
+			refreshHandler := func(appCR appv1.Application, closeRefreshHandler chan struct{}) {
+				for {
+					select {
+					case <-closeRefreshHandler:
+						return
+					default:
+					}
+
+					err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&appCR), &appCR)
+						if err != nil {
+							if apierr.IsNotFound(err) {
+								return nil
+							}
+							return err
+						}
+
+						if _, ok := appCR.Annotations[appv1.AnnotationKeyRefresh]; ok {
+							delete(appCR.Annotations, appv1.AnnotationKeyRefresh)
+							return k8sClient.Update(ctx, &appCR)
+						}
+						return nil
+					})
+					Expect(err).To(BeNil())
```

Comment:
Since this function is called within a goroutine, I think you need  `defer GinkgoRecover()` at the top for Ginkgo to catch this.